File: benchmarks/suite/fibonacci/bench.cpp
#include <cstdio>

int fib(int n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main() {
    int result = fib(35);
    printf("fib(35) = %d\n", result);
    return 0;
}

File: benchmarks/suite/fibonacci/bench.py
"""OBO Benchmark: Fibonacci (recursive)
Tests: function call overhead, recursion depth, integer arithmetic"""

import sys
sys.setrecursionlimit(100000)

def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)

result = fib(35)
print("fib(35) = " + str(result))
